src/main/kotlin/game/blackjack/domain/HandCards.kt

```diff
@@ -0,0 +1,34 @@
+package game.blackjack.domain
+
+class HandCards {
+    private val handCards: MutableList<Card> = mutableListOf()
+    val size: Int get() = handCards.size
+
+    fun add(card: Card) {
+        handCards.add(card)
+    }
+
+    fun add(cards: List<Card>) {
+        handCards.addAll(cards)
+    }
+
+    fun getCurrentScore(): Int {
+        val score = handCards.sumOf { it.number.number }
+
+        return adjustAceScore(score)
+    }
+
+    private fun adjustAceScore(originScore: Int): Int {
+        var score = originScore
+
+        handCards.filter { it.number == CardNumber.ACE }
+            .filter { score + 10 <= Blackjack.WINNING_SCORE }
+            .forEach { _ -> score += 10 }
+
+        return score
+    }
+
+    override fun toString(): String {
+        return handCards.map { it.toString() }.joinToString(", ")
+    }
```

Comment:
다음과 같이 줄여볼 수 있겠네요.
하지만 전반적으로 toString을 통해 출력을 도메인에서 관리하고 있는데,
출력 양식에 대한 역할은 View가 가지고 있어야 한다고 생각해요.

```suggestion
override fun toString(): String = handCards.joinToString(", ") { it.toString() }
```

Reply:
저도 출력 양식에 대한 역할을 View가 가지고 있어야 한다는 의견에는 공감합니다. 🤔
하지만 만약 View 객체에 역할을 위임하려면, 굳이 public으로 빼지 않고 private으로도 사용할 수 있는 상태들을 출력 형태 때문에 public으로 오픈한다는 것에 고민이 됐었는데요. 😅
물론 백킹 프러퍼티로 불변 상태를 반환할 수는 있지만, 어떤게 더 코틀린 스럽게 짜는 방법인지 알려주실 수 있으신가요?

Reply:
@giibeom 
질문주신 것에 대해 제 생각 먼저 공유드리면,
[저는 도메인 내부의 멤버 변수를 그대로 오픈하는 걸 좋아하지 않고, 적절한 메시지(메서드)를 만들어 주는 것을 선호합니다.]

위에 size와 같은 경우도 마찬가지예요.
일급 컬렉션으로 유지할 수 있는 클래스를 또 다른 멤버 변수를 만들다보니,
value class로 변환이나 받은 List로 위임을 할 수도 없는 형태가 되었으니 저걸 메서드로 빼는 게 장점이 더 많다고 생각해요.

---

저는 실무에서는 dto와 메서드가 하나 더 나오더라도 목적에 맞게 매핑해서 던져주는 걸 선호합니다.
현재는 dto까지 만들 필요없이 backing property로 핸드 카드를 복사해서 주면 해당 데이터로 뷰에서 separator를 &로 하든, 다른 뷰의 형태로 가공하든 더 자유로운 형태가 가능하지 않을까요?

이건 코틀린스럽게라기보단 데이터 관리의 주체를 어디서 가져가느냐에 대한 내용이라고 생각하는데,
UI의 핸들링을 서버에서 가져가는 서버 드리븐에서도 보편적으로 특정 메타데이터에 대해서만 수정을 해서 화면을 컨트롤하고, UI에서 해당 메타데이터에 대한 정의만 내려주는 역할을 합니다.

하지만 현재 형태는 정의가 아닌 값에 대한 노출을 모두 도메인에서 제어하고 있는 역할이기에 도메인과 뷰가 강결합되어 있어 말씀드린 거예요.

기범님께서는 위의 내용보다 toString으로 얻는 게 더 많은 거 같다면 유지하셔도 됩니다. 😊